### ISSUE TITLE
`strictOptionalProperties` should not affect comparability

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#44420]
```ts
declare let x: { x?: number };
declare let y: { x: undefined }
if (x === y) {
    // ...
}
```

**Expected**: no error
**Actual**:

```
if (x === y) {
    ~~~~~~~
This condition will always return 'false' since the types '{ x?: number; }' and '{ x: undefined; }' have no overlap.
```

**Comment by DanielRosenwasser** [comment]
Hmm, on second thought, that's a weird case. This only happens when you have a required property whose type is `undefined`.
